src/base/utils/environment_p.cpp

```diff
@@ -280,3 +280,44 @@ Utils::Environment::Impl::CustomProfile::applicationSettings(const QString &name
     const QString settingsFileName {QDir(configLocation()).absoluteFilePath(name + QLatin1String(".conf"))};
     return SettingsPtr(new QSettings(settingsFileName, QSettings::IniFormat));
 }
+
+QString Utils::Environment::Impl::NoConvertConverter::fromPortablePath(const QString &portablePath) const
+{
+    return portablePath;
+}
+
+QString Utils::Environment::Impl::NoConvertConverter::toPortablePath(const QString &path) const
+{
+    return path;
+}
+
+Utils::Environment::Impl::Converter::Converter(const QString &basePath)
+    : m_baseDir {basePath}
+{
+    m_baseDir.makeAbsolute();
+}
+
+QString Utils::Environment::Impl::Converter::toPortablePath(const QString &path) const
+{
+    if (path.isEmpty() || m_baseDir.path().isEmpty())
+        return path;
+
+#ifdef Q_OS_WIN
+    if (QDir::isAbsolutePath(path)) {
+        QChar driveLeter = path[0].toUpper();
+        QChar baseDriveLetter = m_baseDir.path()[0].toUpper();
+        bool onSameDrive = (driveLeter.category() == QChar::Letter_Uppercase) && (driveLeter == baseDriveLetter);
```

Comment:
I guess you imply `--portable` switch set. Things will not work in this scenario (obviously). That is why we split the portable mode onto two options.